comms/core/src/net_address/mutliaddresses_with_stats.rs

```diff
@@ -129,8 +109,65 @@ impl MultiaddressesWithStats {
         self.addresses.iter().map(|addr| addr.address())
     }
 
-    pub fn merge(&mut self, other: &MultiaddressesWithStats) {
-        for addr in &other.addresses {
+    /// Merges another set of addresses into this set, preserving usage stats where possible.
+    ///
+    /// Merge logic:
+    /// - If the incoming set has a strictly newer signed claim, all non-config addresses are replaced by the incoming
+    ///   set.
+    /// - If the incoming set is older, only usage stats for known addresses are updated.
+    /// - If neither set has a signed claim or timestamps are equal, addresses are merged individually.
+    ///
+    /// After merging, addresses are sorted by quality and truncated to the maximum allowed.
```

Comment:
2 questions here, quick one whould we ever get non-signed addresses? And is this not a security issue?

And then the other one, why are we filtering on the config also here?
Zooming out, I get its a manually inserted address, but if the node does not even advertise that addresses a connectable address, why keep it in there if its not dialable?